src/utils/notification.js

```diff
@@ -74,18 +75,30 @@ export class LoadingNotification extends Notification {
 }
 
 export class DeeplinkNotification extends Notification {
-  constructor(body, title, reopen = () => {}, uid, uri, acchash) {
+  constructor(body, title, reopen = () => {}, uid, uri, acchash, extraParams = null) {
```

Comment:
The extraParams property is general (not easy to see what goes here) and seems unnecessary, please extrapolate the properties within it that seem to be put there for the Login request (iaddr, fqn, chain) from the uri in the deeplinknotification, and make fromService a property of deeplink notification. Alternatively, I might be missing something so we can discuss a different solution.